packages/loader/container-loader/src/connectionStateHandler.ts

```diff
@@ -127,6 +128,10 @@ export class ConnectionStateHandler {
         }
     }
 
+    private transitionToConnectedStateWhenCaughtUp() {
+        this.handler.onCaughUpToKnownOps(() => this.setConnectionState(ConnectionState.Connected));
```

Comment:
I do not think this is correct flow. In a sense that it would work only if we do not lose connection while processing ops.
If we do, we have to abandon this process and start over.